src/contactgroup.cc

```diff
@@ -0,0 +1,150 @@
+#include "com/centreon/engine/contact.hh"
+#include "com/centreon/engine/contactgroup.hh"
+#include "com/centreon/engine/error.hh"
+#include "com/centreon/engine/globals.hh"
+#include "com/centreon/engine/logging/logger.hh"
+#include "com/centreon/engine/string.hh"
+#include "find.hh"
+
+using namespace com::centreon;
+using namespace com::centreon::engine;
+using namespace com::centreon::engine::logging;
+
+/**************************************
+*                                     *
+*           Public Methods            *
+*                                     *
+**************************************/
+
+/**
+ * Constructor.
+ */
+contactgroup::contactgroup() {}
+
+/**
+ *  Constructor from a configuration contactgroup
+ *
+ * @param obj Configuration contactgroup
+ */
+contactgroup::contactgroup(configuration::contactgroup const& obj)
+  : _alias(obj.alias().empty() ? obj.contactgroup_name() : obj.alias()),
+    _name(obj.contactgroup_name()) {
+  // Make sure we have the data we need.
+  if (_name.empty())
+    throw (engine_error() << "contactgroup: Contact group name is empty");
+
+  // Notify event broker.
+  timeval tv(get_broker_timestamp(NULL));
+  broker_group(
+    NEBTYPE_CONTACTGROUP_ADD,
+    NEBFLAG_NONE,
+    NEBATTR_NONE,
+    this,
+    &tv);
+}
+
+/**
+ * Assignment operator.
+ *
+ * @param[in] other Object to copy.
+ *
+ * @return This object
+ */
+contactgroup& contactgroup::operator=(contactgroup const& other) {
+  _alias = other._alias;
+  _members = other._members;
+  _name = other._name;
+  return (*this);
+}
+
+/**
+ * Destructor.
+ */
+contactgroup::~contactgroup() {}
+
+std::string const& contactgroup::get_name() const {
+  return _name;
+}
+
+void contactgroup::add_member(contact* cntct) {
+  _members[cntct->get_name()] = cntct;
+
+  timeval tv(get_broker_timestamp(NULL));
+  broker_group(
+    NEBTYPE_CONTACTGROUP_ADD,
+    NEBFLAG_NONE,
+    NEBATTR_NONE,
+    this,
+    &tv);
+  return ;
+}
+
+void contactgroup::clear_members() {
+  _members.clear();
+  return ;
+}
+
+std::unordered_map<std::string, contact*> const& contactgroup::get_members() const {
+  return (_members);
+}
+
+bool contactgroup::contains_illegal_object_chars() const {
+  if (_name.empty() || !illegal_object_chars)
+    return false;
+  return (_name.find(illegal_object_chars) != std::string::npos);
+}
+
+std::string const& contactgroup::get_alias() const {
+  return _alias;
+}
+
+void contactgroup::set_alias(std::string const& alias) {
+  _alias = alias;
+  return ;
+}
+
+bool contactgroup::has_member(std::string const& name) const {
+  return (_members.find(name) != _members.end());
+}
+
+std::ostream& operator<<(
+                std::ostream& os,
+                contactgroup_map const& obj)
+{
+  for (contactgroup_map::const_iterator
+         it(obj.begin()), end(obj.end());
+       it != end;
+       ++it) {
+    os << string::chkstr(static_cast<const char *>(it->first.c_str()));
```

Comment:
The same as previously, we can avoid chkstr...

Reply:
done